apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionScopeTests.cs

```diff
@@ -365,6 +365,36 @@ public void MultiTableWrite()
                 }
             }
         }
+        [Fact]
+        public async Task ParallelWriteAsync()
+        {
+            string[] keys = new string[] { IdGenerator.FromGuid(), IdGenerator.FromGuid(), IdGenerator.FromGuid() };
+            await RetryHelpers.ExecuteWithRetryAsync(async () =>
+            {
+                using var scope = new TransactionScope();
+                using var connection = _fixture.GetConnection();
+                await connection.OpenAsync();
+
+                await Task.WhenAll(keys.Select(key =>
+                {
+                    using var cmd = connection.CreateInsertCommand(_fixture.TableName);
+                    cmd.Parameters.Add("K", SpannerDbType.String).Value = key;
+                    cmd.Parameters.Add("StringValue", SpannerDbType.String).Value = "text";
+                    return cmd.ExecuteNonQueryAsync();
+                }));
+                scope.Complete();
+            });
+
+            // Read the inserted values.
+            using var connection = _fixture.GetConnection();
+            using var command = connection.CreateSelectCommand($"SELECT COUNT(*) AS C FROM {_fixture.TableName} WHERE K IN UNNEST(@Keys)");
+            command.Parameters.Add("Keys", SpannerDbType.ArrayOf(SpannerDbType.String)).Value = keys;
+            using var reader = command.ExecuteReader();
+
+            Assert.True(reader.Read());
+            Assert.Equal(keys.Length, reader.GetInt32(0));
+            Assert.False(reader.Read());
```

Comment:
nit: I would remove this one, your select is a COUNT and if that returns more than one row, that's a problem with the reader, not with this test.

Reply:
Removed.